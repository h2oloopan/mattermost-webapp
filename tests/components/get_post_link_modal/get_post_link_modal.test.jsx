// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import GetPostLinkModal from 'components/get_post_link_modal/get_post_link_modal.jsx';
import GetLinkModal from 'components/get_link_modal.jsx';

describe('components/GetPostLinkModal', () => {
    test('should match snapshot with currentTeamUrl passed in', () => {
        const wrapper = shallow(
            <GetPostLinkModal
                currentTeamUrl='http://current-team-url'
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should call hide on GetLinkModal\'s onHide', () => {
        const wrapper = shallow(
            <GetPostLinkModal
                currentTeamUrl='http://current-team-url'
            />
        );

        wrapper.find(GetLinkModal).first().props().onHide();
        expect(wrapper.state('show')).toBe(false);
    })
});
