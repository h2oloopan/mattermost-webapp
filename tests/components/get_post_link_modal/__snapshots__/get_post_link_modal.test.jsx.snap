// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GetPostLinkModal should match snapshot with currentTeamUrl passed in 1`] = `
<GetLinkModal
  helpText="The link below allows authorized users to see your post."
  link="http://current-team-url/pl/undefined"
  onHide={[Function]}
  show={false}
  title="Copy Permalink"
/>
`;
